### ISSUE TITLE
Cookie is set to empty string when language is not specified (also cookie value injection vulnerability)

### ISSUE THREAD

**Issue by ArnauAregall** [tinganho/express-request-language#5]
If you perform a GET request to `/languages/` (notice ending slash and that 'language' param is empty), cookie is set as an empty string.

``` bash
$ curl -s -D - http://localhost:3000/languages/ | grep -i 'Set-Cookie'
Set-Cookie: language=; Path=/
```

This even applies when you set a language that is not configured in `languages` option:

``` javascript
var requestLanguage = require('express-request-language');
var cookieParser = require('cookie-parser');
var express = require('express');
var app = express();

app.use(cookieParser());
app.use(requestLanguage({
  languages: ['en', 'es'],
  cookie: {
    name: 'language',
    url: '/languages/{language}'
  }
}));
```

...and perform such a request:

``` bash
$ curl -s -D - http://localhost:3000/languages/en | grep -i 'Set-Cookie'
Set-Cookie: language=en; Path=/

$ curl -s -D - http://localhost:3000/languages/notSupportedLanguage | grep -i 'Set-Cookie'
Set-Cookie: language=notSupportedLanguage; Path=/
```

In my opinion, middleware should validate that the received language is actually supported (in the `languages` array) before setting the cookie.

**Comment by tinganho** [comment]
@ArnauAregall you are right. We should check that the language is supported.

**Comment by tinganho** [comment]
This is now fixed in '@1.1.1'. Please try it out. It 404 if the language is not supported.

**Comment by ArnauAregall** [comment]
@tinganho thanks for reviewing it.
I created PR #6 to make the 404 response null safe when there's no match.
